openspoor/visualisations/spoorkaart.py

```diff
@@ -0,0 +1,222 @@
+        spoorkaart
+        :param save_name: An Optional Path where the final .html file could be written to
+        :param makesmaller: A boolean for decreasing the size of the final plot. Designed for Databricks Notebooks
+        :return: A SpoorKaart object
+        """
+        self.makesmaller = makesmaller
+        super().__init__(location=[52, 5], zoom_start=8, max_zoom=30, max_native_zoom=30, tiles=None, **kwargs)
+        self.save_name = save_name
+        self._add_luchtfoto()
+
+        for obj in objects:
+            assert issubclass(type(obj), PlotObject), f'Unable to plot {obj}, not defined as plottable object'
+            obj.add_to(self)
+
+    def _add_luchtfoto(self) -> None:
+        """
+        Add the most recent ProRail luchtfoto to the map.
+
+        :return: None
+        """
+        fg = folium.FeatureGroup(name=f"luchtfoto", max_zoom=30, max_native_zoom=30)
+        folium.WmsTileLayer(url='https://luchtfoto.prorail.nl/erdas-iws/ogc/wms/Luchtfoto', layers='meest_recent',
+                            transparent=True, overlay=False,
+                            maxZoom=30, maxNativeZoom=30).add_to(fg)
+        folium.TileLayer('openstreetmap', transparent=True, opacity=0.2).add_to(fg)
+        self.add_child(fg)
+
+    def __enter__(self):
+        return self
+
+    def _fix_zoom(self) -> None:
+        """
+        Set the zoom level so all of the plotted objects fit neatly within the default shown window.
+
+        :return: None
+        """
+        bboxes = []
+
+        for _, item in self._children.items():
+            for _, subitem in item._children.items():
+                try:
+                    bboxes.append(
+                        subitem.__dict__['data']['features'][0]['bbox'][::-1])  # Reverse lat and long ordering here
+                except:
+                    pass
+            try:
+                bboxes.append(item.location * 2)  # Change order
+            except:
+                pass
+
+        if bboxes:  # Fit only if there are some items to show
+            bounds = [min(map(lambda x: x[i], bboxes)) if i < 2 else max(map(lambda x: x[i], bboxes)) for i in range(4)]
+            self.fit_bounds([bounds[:2], bounds[2:]])
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """
+        End the context manager, keeping track of any exceptions that may have occurred.
+        :return: A SpoorKaart object
+        """
+        if exc_type is not None:
+            raise exc_type(exc_val)
+        self._fix_zoom()
+        if self.save_name is not None:
+            print(f'Saving as {self.save_name}')
+            self.save(self.save_name)
+        if self.makesmaller:
+            self.superfigure = folium.Figure(1000, 400)
+            self.add_to(self.superfigure)
+            return self.superfigure
+        else:
+            return self
+
+
+class PlottingDataFrame(pd.DataFrame, PlotObject):
```

Comment:
The idea was that multiple layers could be plotted on the same map. In a similar fashion I wanted to add support for geopandas linestring, and in the future possibly also for areas. I think this can still be done by setting the map, and then defining a plot_on_map method for every object that would be plotted on it.